internal/controller/config/config.go

```diff
@@ -179,3 +190,24 @@ func (cm *ConfigManager) RegisterCallback(f func(config GlobalConfig)) {
 
 	cm.callbacks = append(cm.callbacks, f)
 }
+
+func (cm *ConfigManager) UpdateUSDEConfig(config *USDEConfig) {
+	cm.usdeConfigLock.Lock()
+	defer cm.usdeConfigLock.Unlock()
+
+	cm.usdeConfig = config
+}
+
+func (cm *ConfigManager) UnsetUSDEConfig() {
+	cm.usdeConfigLock.Lock()
+	defer cm.usdeConfigLock.Unlock()
+
+	cm.usdeConfig = &USDEConfig{}
+}
+
+func (cm *ConfigManager) GetUSDEConfig() *USDEConfig {
+	cm.usdeConfigLock.Lock()
+	defer cm.usdeConfigLock.Unlock()
+
+	return cm.usdeConfig
```

Comment:
should we return a copy of this since otherwise the caller can refer to the reference while it's being modified, right?

Reply:
is the intention is to see the latest modification anyway?

Reply:
I'm just referring to the fact that we have a lock, but is it effective if the caller has a reference? those values in the reference can change after it calls `GetUSDEConfig()`

Reply:
@afugazzotto I don't understand why we return a copy here. if the copy is used how to detect the runtime changes that happened? unless this is only used for unit test. marking this as change request until the question get clarified.

Reply:
@xdevxy if the config is being used and there is a change, then it would make more sense to return the reference instead of making a copy. However, I see a similar logic [here](https://github.com/numaproj/numaplane/blob/main/internal/controller/config/config.go#L66)? Would we also need to change that? How is the GlobalConfig handling changes when returning a copy?

Reply:
If you're returning a reference, then I don't see any point in having a lock.

The GlobalConfig was purposely written to return a copy - the idea is that the caller may want to get multiple values from the ConfigMap and ensure atomic correctness between them.

Reply:
The call to `GetConfig()` is used periodically in goroutines to get an atomically correct version of the Config.

In addition, the GlobalConfig has a callback capability - this is used to update any other copies of values that are stored. That's what we're doing for the LogLevel.

Reply:
Hey Hao. Is there any need to hold up this PR? When I spoke to you on Friday, it sounded like you just wanted to understand how Tony was going to use the code?

Reply:
yes, sorry was out yesterday. I think it is fine to return a reference, however the read and write to the reference needs to be protected by lock (e.g. rwlock). If we return a copy then like what Julie said we need to use a callback or etc. I'm fine with either model, and I don't want to hold this PR for this. We can do as a follow up.

Reply:
Made changes after last conversation with @xdevxy